src/libsync/owncloudpropagator.h

```diff
@@ -187,34 +188,42 @@ class OWNCLOUDSYNC_EXPORT PropagateDirectory : public PropagatorJob {
     // e.g: create the directory
     QScopedPointer<PropagateItemJob>_firstJob;
 
-    // all the sub files or sub directories.
-    QVector<PropagatorJob *> _subJobs;
+    // all the new and changed files without conflicts.
+    QLinkedList<PropagatorJob *> _syncJobs;
```

Comment:
AFAIK not ideal because of heap fragmentation, but @ogoffart can say more to this.